src/puppetlabs/puppetdb/http/aggregate_event_counts.clj

```diff
@@ -1,49 +1,23 @@
 (ns puppetlabs.puppetdb.http.aggregate-event-counts
-  (:require [puppetlabs.puppetdb.cheshire :as json]
-            [puppetlabs.puppetdb.http.events :refer [validate-distinct-options!]]
-            [puppetlabs.puppetdb.http :as http]
-            [puppetlabs.puppetdb.utils :as utils]
-            [puppetlabs.puppetdb.query-eng :refer [produce-streaming-body]]
-            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params]]
-            [clojure.tools.logging :as log]
+  (:require [puppetlabs.puppetdb.http.query :as http-q]
+            [puppetlabs.puppetdb.middleware :refer [verify-accepts-json]]
             [net.cgrand.moustache :refer [app]]))
 
 (defn routes
-  [version]
-  (app
-    [""]
-    {:get (fn [{:keys [params globals]}]
-            (when (= "puppetdb" (:product-name globals))
-              (log/warn "The aggregate-event-counts endpoint is experimental"
-                        " and may be altered or removed in the future."))
-            (if (utils/hsql? (:scf-read-db globals))
-              (http/json-response
-                {:error "The aggregate-event-counts endpoint does not support HSQLDB."}
-                http/status-not-implemented)
-              (let [{:strs [query summarize_by counts_filter count_by]} params
-                    distinct-options (validate-distinct-options! params)
-                    counts_filter' (json/parse-string counts_filter true)
-                    query-options (-> {:counts_filter counts_filter'
-                                       :count_by count_by
-                                       :summarize_by summarize_by}
-                                      (merge distinct-options))]
-                (log/warn
-                  (str "The aggregate-event-counts endpoint is experimental"
-                       " and may be altered or removed in the future."))
-                (produce-streaming-body
-                  :aggregate-event-counts
-                  version
-                  query
-                  query-options
-                  (:scf-read-db globals)
-                  (:url-prefix globals)))))}))
+  [version optional-handlers]
```

Comment:
It seems odd to me that we flow around `optional-handlers`, when all `query-route` does with it is `comp` it onto its result. Couldn't this be done more cleanly by doing the `comp` on the result of query-route, when it's needed?

(I know we had optional-handlers before; it this is a change for another time, that's fine)